### PR TITLE
Rishabh/enurse integration

### PR DESCRIPTION
**Pull request by cmile-rishabh-kukkar**

### CODE REVIEW

File: src/main/java/com/gulteking/mqttbackendserver/controller/DeviceController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/mqtt-device")
+public class DeviceController {
+    @Autowired
+    DeviceService deviceService;
+
+    @PostMapping("/toTopic")
+    public ResponseEntity<Object> subscribeTopic(@RequestBody MqttTopicSubscribe messagePublishModel) {
```
Comment (akshaymemane) [inline src/main/java/com/gulteking/mqttbackendserver/controller/DeviceController.java]: misleading request type and variable.

File: src/main/java/com/gulteking/mqttbackendserver/controller/DeviceController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        return deviceService.subscribeTopic(messagePublishModel);
+    }
+
+    @PostMapping("/toAllTopic")
+    public ResponseEntity<Object> subscribeAllTopic(@RequestBody Boolean subscribeUnsubscribe) {
```
Comment (akshaymemane) [inline src/main/java/com/gulteking/mqttbackendserver/controller/DeviceController.java]: can we rename input variable to make it more easy to understand

File: src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    @Autowired
+    DeviceRepository deviceRepository;
+
+    @Autowired
+    MqttSubscriberImpl mqttSubscriber;
+
+    @Autowired
+    MqttPublisherImpl mqttPublisher;
+
+    @Override
```
Comment (akshaymemane) [inline src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java:29]: add @Transactional where ever we are doing multiple write operations.

File: src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                    .message("Something went wrong while subscribing Topic!!!")
+                    .status(HttpStatus.BAD_REQUEST)
+                    .statusCode(HttpStatus.BAD_REQUEST.value())
+                    .build());
+        }
+    }
+
+    @Override
+    public ResponseEntity<Object> subscribeTopic(MqttTopicSubscribe messagePublishModel) {
+        Optional<Device> topics = deviceRepository.findById(messagePublishModel.getTopicId());
```
Comment (akshaymemane) [inline src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java]: rename topics to devices. as we are fetching devices using deviceRepository

File: src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java (hunk 3 above)
Comment (cmile-rishabh-kukkar) [reply src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java:29]: we are not writing anything into database .

Comment (akshaymemane) [reply src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java:29]: ok

**Review by akshaymemane — APPROVED** [review]
LGTM.

**Pull request merged**